### PR TITLE
Fix for Process.TestStartTime flakiness.

### PR DESCRIPTION
**Pull request by Priya91**
#2613 

cc @stephentoub

### CODE REVIEW

**Comment by stephentoub** [comment]
Thanks, Priya. Can you elaborate on what the problem was and how this fixes it?

**Comment by Priya91** [comment]
> Can you elaborate on what the problem was and how this fixes it?

Sure. The test was always failing with the Process.StartTime value being greater that the afterTicks value. This could be because:

1. p.Start()
2. long afterTicks = DateTime.Now.Ticks
3. OS actually starts process now
4. Check p.StartTime.Ticks < afterTicks => which will be false.

With the proposed change I am ensuring the Process was actually started by the OS, by checking if there is a running process with p.Id using Process.GetProcessById().

File: src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs [hunk 1]
```diff
@@ -133,8 +133,21 @@ public void TestExitTime()
...
-            Assert.True(p.ExitTime.ToUniversalTime() > timeBeforeProcessStart, "TestExitTime is incorrect.");
+            try
+            {
+                p.Start();
+                Assert.Throws<InvalidOperationException>(() => p.ExitTime);
+                p.Kill();
+                Assert.True(p.WaitForExit(WaitInMS));
+                Assert.True(p.ExitTime.ToUniversalTime() > timeBeforeProcessStart, "TestExitTime is incorrect.");
+            }
+            finally
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs]: What's the purpose of this finally block?  I thought we weren't doing things like this in individual tests and letting the test class' Dispose clean them up.  Does this also need to be done here for some reason?

**Comment by stephentoub** [comment]
One question, otherwise LGTM.

File: src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs (hunk 1 above)
Comment (Priya91) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs]: Yes, we don't need this. CreateProcessInfinite adds the process to the list of processes to be disposed.

**Pull request merged**